lib/configuration/configuration.ts

```diff
@@ -2,84 +2,85 @@ import { BehaviorSubject } from "rxjs";
 import { config } from "@mtfh/common/lib/config";
 import { axiosInstance } from "@mtfh/common/lib/http";
 
-export type Configuration = {
-  type: "MMH" | "Common";
-  configuration: Record<string, string>;
-  featureToggles: Record<string, boolean>;
+type Configuration = {
+  configuration: Record<string, string | undefined>;
+  featureToggles: Record<string, boolean | undefined>;
 };
+type ConfigCollection = Record<string, Configuration>;
 
-const initialFeatureToggles = {
+type ConfigResponse = Array<Configuration & { type: string }>;
+
+const initialConfiguration: ConfigCollection = {
   MMH: {
-    Test: false,
-    CreateTenure: false,
-    EnhancedPersonComments: false,
-    WarningComponents: false,
-    Stepper: false,
+    configuration: {
+      TestConfig: "",
+    },
+    featureToggles: {
+      Test: false,
+      CreateTenure: false,
+      EnhancedPersonComments: false,
+      WarningComponents: false,
+      Stepper: false,
+    },
   },
 };
 
-export const featureToggleStore = new BehaviorSubject(initialFeatureToggles);
-
-type PathImpl<T, Key extends keyof T> = Key extends string
-  ? T[Key] extends Record<string, any>
-    ?
-        | `${Key}.${PathImpl<T[Key], Exclude<keyof T[Key], keyof any[]>> & string}`
-        | `${Key}.${Exclude<keyof T[Key], keyof any[]> & string}`
-    : never
-  : never;
-
-type Path<T> = PathImpl<T, keyof T>;
-
-export type FeatureTogglePaths = Path<typeof initialFeatureToggles>;
-
-export const getConfiguration = async (): Promise<void> => {
+export const hydrateConfiguration = () => {
   try {
     const features = JSON.parse(
       window.localStorage.getItem("features") || "",
-    ) as typeof initialFeatureToggles;
+    ) as typeof initialConfiguration;
 
     if (typeof features === "object") {
-      featureToggleStore.next(features);
-    } else {
-      throw new Error("Invalid feature store in local storage");
+      return features;
     }
+    throw new Error("Invalid feature store in local storage");
   } catch (e) {
     if (localStorage.getItem("features")) {
       window.localStorage.removeItem("features");
     }
   }
+  return {};
+};
 
+export const $configuration = new BehaviorSubject(hydrateConfiguration());
+
+export const getConfiguration = async (): Promise<void> => {
```

Comment:
this is the real hydrateConfiguration to my eyes
The other one is the one from which you get it